### PR TITLE
fix: [PL-32654]: Added limits for connectors based on license

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
## Describe your changes
This pr adds limits to connectors count based on license of the account. For free and team editions, it is limited to 100 connectors per account. 

Testing:
1. Tested locally with free and enterprise edition and verified the max count.
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47794)

### CODE REVIEW

File: 925-enforcement-service/src/main/resources/features/platform_features.yml [hunk 1]
```diff
@@ -122,6 +122,23 @@ features:
...
+      FREE:
+        restrictionType: STATIC_LIMIT
+        limit: 100
+        clientName: ngmanager
+      TEAM:
+        restrictionType: AVAILABILITY
+        enabled: true
+      ENTERPRISE:
+        restrictionType: AVAILABILITY
+        enabled: true
```
Comment (nishantsaini20) [inline 925-enforcement-service/src/main/resources/features/platform_features.yml:141]: Can tests be added to validate behaviour for this addition on connector creation?

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]

**Comment by VikasMaddukuriHarness** [comment]
trigger uts

**Comment by VikasMaddukuriHarness** [comment]
trigger ut1,ut2,ut3

**Comment by VikasMaddukuriHarness** [comment]
trigger ut7

**Comment by VikasMaddukuriHarness** [comment]
trigger uts

File: 925-enforcement-service/src/main/resources/features/platform_features.yml (hunk 1 above)
Comment (VikasMaddukuriHarness) [reply 925-enforcement-service/src/main/resources/features/platform_features.yml:141]: I don't think we have tests for these. I don't think testing these changes are that necessary. But please let me know if you think we can add. Thanks.

Comment (nishantsaini20) [reply 925-enforcement-service/src/main/resources/features/platform_features.yml:141]: This addition changes the behaviour of connect create api. Therefore IMO we should have tests around it but if this need proper framework then we can have is as part of different PR.

**Review by nishantsaini20 — APPROVED** [review]

**Review by tiwaritejas — APPROVED** [review]
> Reviewed 5 of 10 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47794)* status: 10 file(s) to review.

**Pull request merged**